src/main/java/org/mule/runtime/policy/api/OperationPolicyPointcutParametersFactory.java

```diff
@@ -27,13 +27,24 @@
   boolean supportsOperationIdentifier(ComponentIdentifier operationIdentifier);
 
   /**
-   * Creates an specific {@link PolicyPointcutParameters} for a particular operation identifier by {@code operationIdentifier}.
+   * Creates a specific {@link PolicyPointcutParameters} for a particular operation identifier by {@code operationIdentifier}.
    *
    * @param operation the operation where the policy is being applied.
    * @param operationParameters set of parameters that are going to be used to execute the operation.
    * @return the pointcut parameters.
    */
-  PolicyPointcutParameters createPolicyPointcutParameters(Component operation,
-                                                          Map<String, Object> operationParameters);
+  @Deprecated
+  default PolicyPointcutParameters createPolicyPointcutParameters(Component operation,
+                                                                  Map<String, Object> operationParameters) {
+    return createPolicyPointcutParameters(new OperationPolicyPointcutParametersParameters(operation, operationParameters));
+  };
+
+  /**
+   * Creates a specific {@link PolicyPointcutParameters} using the specified {@link OperationPolicyPointcutParametersParameters}
+   *
+   * @param pointcutParametersParameters parameters used for creating the {@link PolicyPointcutParameters}
+   * @return the pointcut parameters.
+   */
+  PolicyPointcutParameters createPolicyPointcutParameters(OperationPolicyPointcutParametersParameters pointcutParametersParameters);
```

Comment:
I think you need to put a default implementation for this method so the already released version of HTTP connector works but, to avoid making this API ugly you can try calling this method and catch a NoMethodDefFound exception and call the old method instead. Please also test that it works.